all/pom.xml

```diff
@@ -213,5 +214,108 @@
+                        <artifactId>filevault-package-maven-plugin</artifactId>
+                        <inherited>false</inherited>
+                        <executions>
+                            <execution>
+                                <id>generate-metadata-cloud</id>
+                                <goals>
+                                    <goal>generate-metadata</goal>
+                                </goals>
+                                <configuration>
+                                    <classifier>cloud</classifier>
+                                    <name>acs-aem-commons-all-cloud</name>
+                                    <packageType>container</packageType>
+                                    <properties>
+                                        <cloudManagerTarget>all</cloudManagerTarget>
+                                    </properties>
+                                    <embeddeds>
+                                        <embedded>
+                                            <groupId>${project.groupId}</groupId>
+                                            <artifactId>acs-aem-commons-bundle</artifactId>
+                                            <target>/apps/acs-commons/install</target>
+                                            <filter>true</filter>
+                                            <isAllVersionsFilter>true</isAllVersionsFilter>
+                                        </embedded>
+                                        <embedded>
+                                            <groupId>${project.groupId}</groupId>
+                                            <artifactId>acs-aem-commons-bundle-cloud</artifactId>
+                                            <target>/apps/acs-commons/install</target>
+                                        </embedded>
+                                    </embeddeds>
+                                    <subPackages>
+                                        <subPackage>
+                                            <groupId>${project.groupId}</groupId>
+                                            <artifactId>acs-aem-commons-ui.apps</artifactId>
+                                            <filter>true</filter>
+                                            <isAllVersionsFilter>true</isAllVersionsFilter>
+                                        </subPackage>
+                                        <subPackage>
+                                            <groupId>${project.groupId}</groupId>
+                                            <artifactId>acs-aem-commons-ui.content</artifactId>
+                                            <filter>true</filter>
+                                            <isAllVersionsFilter>true</isAllVersionsFilter>
+                                        </subPackage>
+                                    </subPackages>
+                                </configuration>
```

Comment:
please add package dependencies to require the cloud version and similar the SP level for AEM classic.

Reply:
do you have an example of which dependency to use for the cloud version? 

I see the core components content package uses day/cq60/product:cq-platform-content

Reply:
For Cloud it doesn't seem applicable as the cp2fm tool doesn't evaluate it. Just for enforcing the SP10 with Classic it would be nice (and to prevent the classic version from being installed in Cloud).

Reply:
@kwin  I have set a package dependency for both (can't only set it for classic, else it will be taken in the cloud profile as well, unless I make a profile for classic as well and not build anything by default)

The strange thing is, I have to disable the jackrabbit-packagetype validation, it seems that a container is not supposed to have dependencies? Not sure what the reasoning is there

Reply:
I would use `day/cq60/product:cq-content:6.6` to enforce AEM Cloud, the reason why this is only allowed on non-container packages is that container packages are dissolved for AEM Cloud (with https://github.com/apache/sling-org-apache-sling-feature-cpconverter). Therefore please create dedicated apps packages with specific package dependencies.

Reply:
@davidjgonzalez Do you know if there is any means to depend on a specific AEM Cloud version (as minimum requirement) via package dependencies? This would be crucial as sometimes the import-packages versions from bundles are not restrictive enough (e.g. when certain referenced resource types have only been introduced in a certain AEM Cloud version).

Reply:
@kwin  seeing as the package itself will be named acs-aem-commons-cloud, I'd expect this to be restrictive enough. We will also explicitly mention this in the readme. 

If we do want to go for the dedicated ui.apps.cloud route, what would this entail? What would be in that ui.apps package? 

- use the maven resources plugin to copy everything over from the default ui.apps to an ui.apps.cloud? 
- leave it empty for now and just put the package dependency? This would mean that if you install the acs-aem-commons-cloud on on-prem installation, it would still actually install the "all" package and the bundle, just not the ui.apps.cloud + the bundle.cloud fragment.

Reply:
Using `day/cq60/product:cq-content:6.6` would resolve only in AEMaaCS, but on every version of it. But if this the best we can enforce, we should go for it.